### ISSUE TITLE
images from local html displayed in gray

### ISSUE THREAD

**Issue by H4nsie** [wkhtmltopdf/wkhtmltopdf#1930]
When generating a PDF from a local html (facebook) only the FIRST image in an page is displayed correct, all next images as displayed in gray. This is on all Facebook pages.
The local html displays correctly!

To reproduce: (wkhtmltopdf 0.12.1 (with patched qt) )

--- retrieve sample page
wget -nd  -E -H -k -pr https://mbasic.facebook.com/1mpics?v=photos
--- pdf from sample page
wkhtmltopdf --disable-external-links 1mpics\?v\=photos.html photos.pdf

![screen shot 2014-08-27 at 10 57 36](https://cloud.githubusercontent.com/assets/8566538/4057423/933fd59e-2dc8-11e4-8bf7-ffdea88ddae6.png)

**Comment by sevenseacat** [comment]
I've seen this problem on PDFs I generate in my Rails app for years - some images display normally, others appear faded out like this. No idea what causes it.

**Comment by nineohnine** [comment]
Has there been any progress on this issue?  I'm experiencing the same issue.  @H4nsie were you able to find a work around?

**Comment by PhilterPaper** [comment]
Check if the original HTML is displaying most images "grayed out" (nonselectable or _not_ highlighted). It might be a subtle effect that isn't very visible on a browser, but wkHTMLtoPDF is going a bit overboard on it. Are the images in question processed "on the fly" to change them (e.g., add a watermark)? Again, wkHTMLtoPDF might be mishandling this. The key point here is to determine whether those are "normal" images being displayed, or if they have been "processed" in some way. If normal, is there CSS that wkHTMLtoPDF is mishandling? 

By the way, those photos don't quite look gray scale to me. They look extremely desaturated, but still with a bit of color.

**Comment by nineohnine** [comment]
Agreed they are not grayscale.  As far as I can tell there is nothing css related that could be causing it.  Oddly enough I am also dealing with a FB page. So it's possible they are doing something prior to the image being served.  Although it's not all images only some.

**Comment by PhilterPaper** [comment]
FB only? First image only? Any consistency in which image(s)? You might look at the page source (HTML) for the FB page and see if there's any difference in HTML or CSS between images that are full color and the desaturated images. What happens if you save the page (and images) locally, and process that? Maybe FB has some sort of anti-scraping thing going on to ruin pictures if it thinks a bot is scraping a page?
